flake8_quotes/__init__.py

```diff
@@ -160,37 +122,14 @@ def parse_options(cls, options):
         else:
             cls.config.update({'avoid_escape': True})
 
-    def get_file_contents(self):
-        if self.filename in ('stdin', '-', None):
-            return stdin_get_value().splitlines(True)
-        else:
-            if self.lines:
-                return self.lines
-            else:
-                return readlines(self.filename)
-
-    def run(self):
-        file_contents = self.get_file_contents()
-
-        noqa_line_numbers = self.get_noqa_lines(file_contents)
-        errors = self.get_quotes_errors(file_contents)
-
-        for error in errors:
-            if error.get('line') not in noqa_line_numbers:
-                yield (error.get('line'), error.get('col'), error.get('message'), type(self))
-
-    def get_noqa_lines(self, file_contents):
-        tokens = [Token(t) for t in tokenize.generate_tokens(lambda L=iter(file_contents): next(L))]
-        return [token.start_row
-                for token in tokens
-                if token.type == tokenize.COMMENT and token.string.endswith('noqa')]
-
-    def get_quotes_errors(self, file_contents):
-        tokens = [Token(t) for t in tokenize.generate_tokens(lambda L=iter(file_contents): next(L))]
-        docstring_tokens = get_docstring_tokens(tokens)
-
-        for token in tokens:
+    def __init__(self, logical_line, previous_logical, tokens):
+        self.line = logical_line
+        self.tokens = tokens
+        prev_tokens = tokenize.tokenize(lambda L=iter([previous_logical.encode('utf-8')]): next(L))
```

Comment:
Not your fault as this seems more/less the same as https://github.com/zheller/flake8-quotes/pull/101/files#diff-d7312ce5f3248fb3a2d6e340b7cf0d22L183

```python
tokens = [Token(t) for t in tokenize.generate_tokens(lambda L=iter(file_contents): next(L))]
```

but this is a damn confusing line, we should prob break it up into a more straightforward function

```suggestion
        # Generate `readline()` matching interface for `tokenize.tokenize` and pass it in
        #   https://docs.python.org/3/library/tokenize.html#tokenize.tokenize
        previous_logical_readline_fn = (lambda L=iter([previous_logical.encode('utf-8')]): next(L))
        prev_tokens = tokenize.tokenize(previous_logical_readline_fn)
```

Reply:
Agreed, I was trying to minimize changes.